molecule/provisioners/libvirtprovisioner.py

```
@@ -0,0 +1,923 @@
+                                    'name']))
+                            break
+                        else:
+                            utilities.print_info("\t{}: booting".format(
+                                instance['name']))
+                            dom.create()
+                            if wait:
+                                time.sleep(self._boot_wait)
+                                wait = False
+            if not dom_found:
+                utilities.print_info("\t{}: defining".format(instance['name']))
+                dom = self._libvirt.defineXML(self._build_domain_xml(instance))
+                if GUESTFS:
+                    # TODO: Here is the point at which to modify with libguestfs,
+                    #  i.e. after creating the domain's volume but before booting the domain.
+                    # if any of this instance's interfaces has a key 'address'? or always?
+                    try:
+                        guest = guestfs.GuestFS(python_return_dict=True)
+                        self._manipulate_guest_image(guest, instance)
+                        guest.umount_all()
+                        guest.shutdown()
+                        guest.close()
+                    except Exception, e:
+                        LOG.warning("\nFAILED to manipulate the guest image so could not configure network interfaces: {}".format(e))
+                utilities.print_success("\tCreated instance {}.\n".format(
+                    instance['name']))
+                utilities.print_info("\t{}: booting".format(instance['name']))
+                try:
+                    dom.create()
+                    if wait:
+                        time.sleep(self._boot_wait)
+                        wait = False
+                except libvirt.libvirtError as e:
+                    LOG.error("\nFailed to create/boot {}: {}".format(instance[
+                        'name'], e))
+                    dom.undefine()
```

Shouldn't we re-raise the exception in this case?